test/tap/index-test.js

```diff
@@ -61,22 +63,35 @@ test('log4js rabbitmqAppender', (batch) => {
 
     result.logger('Log event #1');
 
+    t.match(result.fakeRabbitmq.state.url, 'amqps://thing:secret@123.123.123.123:1234/pants');
     t.match(result.fakeRabbitmq.state.params, {
+      protocol: 'amqps',
       hostname: '123.123.123.123',
       port: 1234,
       username: 'thing',
       password: 'secret',
+      locale: 'en_US',
+      frameMax: 0,
+      heartbeat: 60,
+      keepAliveDelay: 0,
+      vhost: '/pants',
       routing_key: 'something',
       exchange: 'exchange_logs',
       mq_type: 'some type',
       durable: true,
-      vhost: '/pants'
+      connection_timeout: 1000,
+      layout: {
+        type: 'pattern',
+        pattern: 'cheese %m'
+      }
     });
 
     result.appender.shutdown(() => {
-      t.equal(result.fakeRabbitmq.state.msgs.length, 1, 'should be one message only');
-      t.equal(result.fakeRabbitmq.state.msgs[0].toString(), 'cheese %m');
-      t.end();
+      setTimeout(() => {
```

Comment:
If you're talking to a fake RabbitMQ you shouldn't need a setTimeout, because it won't be doing anything asynchronous.